torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -22,9 +22,16 @@ class ProfileRegistry {
     return &profile_registry_;
   }
 
-  void registerProfileNode(const std::function<bool(const Node*)>& func) {
+  size_t registerProfileNode(const std::function<bool(const Node*)>& func) {
     std::lock_guard<std::mutex> guard(mutex_);
-    registry_funcs_.push_back(func);
+    const auto key = registry_funcs_.size();
```

Comment:
This is somehow dangerous to use the number of the dictionary as key. Suppose I call unregister and then register once, then the last function will be overridden.

Reply:
Thanks, I have come up with saner version which keeps increasing counter in a member field.